ansible/roles/test/tasks/link_flap/link_flap_helper.yml

```diff
@@ -5,44 +5,43 @@
 - block:
     - set_fact:
         interface: "{{item}}"
-    
+
     - debug: msg={{interface}}
-    
+
     - set_fact:
         peer_device: "{{neighbors[interface]['peerdevice']}}"
         neighbor_interface: "{{neighbors[interface]['peerport']}}"
-    
+
     - conn_graph_facts: host={{ peer_device }}
       connection: local
-    
+
     - set_fact:
         peer_host: "{{device_info['mgmtip']}}"
         peer_hwsku: "{{device_info['HwSku']}}"
-    
+
     - set_fact:
         intfs_to_exclude: "{{interface}}"
-    
+
     - name: Shut down neighbor interface {{neighbor_interface}} on {{peer_host}}
       action: apswitch template=neighbor_interface_shut_single.j2
       args:
         host: "{{peer_host}}"
         login: "{{switch_login[hwsku_map[peer_hwsku]]}}"
       connection: switch
-    
+
     - pause:
         seconds: 20
-    
-    - interface_facts:
-    
+
+    - interface_facts: all_ports={{minigraph_ports}} except_port={{intfs_to_exclude}}
```

Comment:
the intfs_to_exclude is the one that is down? then why not change except_port to down_port and then you can verify the down port also using interface_facts?

Reply:
I didn't do that for 2 reasons:

1. if I include the 'down_port' in the return list, that complicates the test for up ports. If not, then I have to return another list for expected down but up ports.
2. testing of intfs_to_exclude in ansible is only one test and it takes the same amount of time of testing if a returned list is empty.